MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp

```diff
@@ -1536,11 +1523,30 @@ void EnggDiffractionViewQtGUI::setDefaultBank(
     if (combo_data > -1) {
       setBankIdComboBox(combo_data);
     } else {
-      setfittingRunNo(selectedFile);
+      setFittingRunNo(selectedFile);
+    }
+  }
+  // check if the vector is not empty so that the first directory
+  // can be assigned to text-field when number is given
+  else if (!m_fitting_runno_dir_vec.empty()) {
+    auto firstDir = m_fitting_runno_dir_vec.at(0);
+    auto intialDir = QString::fromStdString(firstDir);
+    setFittingRunNo(intialDir);
+  }
+  // if nothing found related to text-field input
+  else if (!getFittingRunNo().empty())
+    setFittingRunNo(selectedFile);
```

Comment:
All these ifs ideally belong in the presenter. `get/setFittingRunNo()` and other functions are in the `IView` interface so they can be used from the presenter (and they're being used in other methods). The view should not make these decisions, just offer the `set/getFittingRunNo()`.